packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -219,38 +239,73 @@ export default class GoogleCalendarService implements Calendar {
       "primary";
 
     try {
-      const event = await calendar.events.insert({
-        calendarId: selectedCalendar,
-        requestBody: payload,
-        conferenceDataVersion: 1,
-        sendUpdates: "none",
-      });
+      let event;
+      let recurringEventId = null;
+      if (calEventRaw.existingRecurringEvent) {
+        recurringEventId = calEventRaw.existingRecurringEvent.recurringEventId;
+        const recurringEventInstances = await calendar.events.instances({
+          calendarId: selectedCalendar,
+          eventId: calEventRaw.existingRecurringEvent.recurringEventId,
+        });
+        event = recurringEventInstances.data.items![calEventRaw.existingRecurringEvent.eventIndex];
+        await calendar.events.patch({
+          calendarId: selectedCalendar,
+          eventId: event.id || "",
+          requestBody: {
+            description: getRichDescription({
+              ...calEventRaw,
+            }),
+          },
+        });
+      } else {
+        const eventResponse = await calendar.events.insert({
+          calendarId: selectedCalendar,
+          requestBody: payload,
+          conferenceDataVersion: 1,
+          sendUpdates: "none",
+        });
+        event = eventResponse.data;
+        if (event.recurrence) {
+          if (event.recurrence.length > 0) {
+            recurringEventId = event.id;
+            const recurringEventInstances = await calendar.events.instances({
+              calendarId: selectedCalendar,
+              eventId: recurringEventId || "",
+            });
+
+            if (recurringEventInstances.data.items) {
+              event = recurringEventInstances.data.items[0];
```

Comment:
Instead of just storing the first occurence we should also plan to store the recurring event ID, so that it can be cleanly deleted from Google Calendar using that ID. It could be a later improvement though.